core/trino-main/src/main/java/io/trino/sql/planner/OptimizerConfig.java

```diff
@@ -471,18 +482,36 @@ public OptimizerConfig setOptimizeMetadataQueries(boolean optimizeMetadataQuerie
         return this;
     }
 
-    public boolean isUseMarkDistinct()
+    @Deprecated
+    @Nullable
+    public Boolean isUseMarkDistinct()
     {
         return useMarkDistinct;
     }
 
+    @Deprecated
     @Config("optimizer.use-mark-distinct")
-    public OptimizerConfig setUseMarkDistinct(boolean value)
+    @LegacyConfig(value = "optimizer.use-mark-distinct", replacedBy = "optimizer.mark-distinct-strategy")
```

Comment:
@raunaqmorarka `@LegacyConfig` causes:
```
Error:  io.trino.TestJdbcResultSetCompatibilityOldDriver.setupServer  Time elapsed: 1.726 s  <<< FAILURE!
io.airlift.bootstrap.ApplicationConfigurationException: 
Configuration errors:

1) Error: @Config method [public OptimizerConfig OptimizerConfig.setUseMarkDistinct(Boolean)] has annotation claiming to be replaced by another property ('optimizer.mark-distinct-strategy')

2) Error: @Config property name 'optimizer.use-mark-distinct' appears in @LegacyConfig annotation for method [public OptimizerConfig OptimizerConfig.setUseMarkDistinct(Boolean)]

```
https://github.com/trinodb/trino/actions/runs/4292550494/jobs/7479120211

Reply:
This is because  `@LegacyConfig` does not mix with `@Config`. Fixed